fca/src/main/java/org/nmdp/ngs/fca/CompleteLattice.java

```diff
@@ -0,0 +1,387 @@
+/*
+
+    ngs-fca  Formal concept analysis for genomics.
+    Copyright (c) 2014-2015 National Marrow Donor Program (NMDP)
+
+    This library is free software; you can redistribute it and/or modify it
+    under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation; either version 3 of the License, or (at
+    your option) any later version.
+
+    This library is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; with out even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library;  if not, write to the Free Software Foundation,
+    Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA.
+
+    > http://www.gnu.org/licenses/lgpl.html
+
+*/
+package org.nmdp.ngs.fca;
+
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import com.tinkerpop.blueprints.Graph;
+import com.tinkerpop.blueprints.Edge;
+import com.tinkerpop.blueprints.Vertex;
+import com.tinkerpop.blueprints.Direction;
+import com.tinkerpop.blueprints.impls.tg.TinkerGraph;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+
+/**
+ * Complete Lattice.
+ * @param <E> element type
+ * @see <a href="https://en.wikipedia.org/wiki/Complete_lattice">complete
+ * lattice </a>
+ */
+public abstract class CompleteLattice<E extends PartiallyOrdered> implements Lattice<E> {
```

Comment:
Think about how this abstract class might be extended, and make final anything you wouldn't want subclasses to mess with